src/main/java/com/google/sps/utility/FreeTimeUtility.java

```diff
@@ -0,0 +1,174 @@
+   * accounting for the work period in there. All time periods are in UTC timezone. TODO: Refactor
+   * the logic of this function to account for edge cases like multiple events being created at the
+   * same time. (Issue #104)
+   *
+   * @param startTime parameter of type long that gives the Unix epoch time of the start of the
+   *     event
+   * @param endTime parameter of type long that gives the Unix epoch time of the end of the event
+   */
+  public void addEvent(long startTime, long endTime) {
+    long numDaysBegin = startTime / numMillisecondsDay;
+    long numDaysEnd = endTime / numMillisecondsDay;
+    long beginDay = (numDaysBegin + UnixEpochDayShift) % 7;
+    long endDay = (numDaysEnd + UnixEpochDayShift) % 7;
+    long startHour = startTime % numMillisecondsDay;
+    long endHour = endTime % numMillisecondsDay;
+    int indexBegin = (int) (beginDay - startDay);
+    int indexEnd = (int) (endDay - startDay);
+    for (int index = indexBegin; index <= Math.min(5, indexEnd); index++) {
+      if (index == indexBegin && index == indexEnd) {
+        setSameDayEvent(index, startHour, endHour);
+      } else if (index == indexBegin) {
+        setStartDayEvent(index, startHour);
+      } else if (index == indexEnd) {
+        setEndDayEvent(index, endHour);
+      } else {
+        setAllDayEvent(index);
+      }
+    }
+  }
+
+  private void setSameDayEvent(int index, long startHour, long endHour) {
+    if (startHour < personalBegin && endHour >= personalEnd) {
+      // Case where we have an all day event
+      setAllDayEvent(index);
+    } else if (startHour < personalBegin && endHour < personalEnd) {
+      // Case where the event starts before the personal time
+      setEndDayEvent(index, endHour);
+    } else if (startHour >= personalBegin && endHour >= personalEnd) {
+      // Case where the event ends after the personal time
+      setStartDayEvent(index, startHour);
+    } else {
+      // Case where the entire event is within the personal and work time
+      long morningStart = Math.min(workBegin, startHour);
+      long morningEnd = Math.min(workBegin, endHour);
+      long eveningStart = Math.max(workEnd, startHour);
+      long eveningEnd = Math.max(workEnd, endHour);
+      long duration = endHour - startHour;
+      long personalDuration = (morningEnd - morningStart) + (eveningEnd - eveningStart);
+      long workDuration = duration - personalDuration;
+      personalHoursPerDay.set(
+          index, Math.max(personalHoursPerDay.get(index) - personalDuration, 0));
+      workHoursPerDay.set(index, Math.max(workHoursPerDay.get(index) - workDuration, 0));
+    }
+  }
+
+  private void setAllDayEvent(int index) {
+    personalHoursPerDay.set(index, (long) 0);
+    workHoursPerDay.set(index, (long) 0);
+  }
+
+  private void setStartDayEvent(int index, long startHour) {
+    if (startHour < personalBegin) {
+      // This case is similar as an all day event
```

Comment:
This was because the hours allocated for this case would be beginning at the start of free time to the end of the day. The refactoring makes this a deprecated function however.